src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -2911,10 +2911,10 @@ let eta_expand (env:Env.env) (t:term) : term =
         let formals, _tres = U.arrow_formals (SS.subst' s u.ctx_uvar_typ) in
         if List.length formals = List.length args
         then t
-        else let _, ty, _ = env.type_of ({env with lax=true; use_bv_sorts=true; expected_typ=None}) t in
+        else let _, ty, _ = env.typeof_tot_or_gtot_term ({env with lax=true; use_bv_sorts=true; expected_typ=None}) t true in
```

Comment:
I wonder if it would be more readable if the interface provided typeof_tot_term and type_of_gtot_term, rather than a single function whose behavior is toggled with a flag. The `true` at the end is a bit opaque to read

Reply:
Sounds good to me, will change.